File: npm-package/uninstall.js
// SPDX-License-Identifier: MPL-2.0

const fs = require('fs');
const path = require('path');

const BIN_DIR = path.join(__dirname, 'bin');
const BINARY_PATH = path.join(BIN_DIR, 'shaper');

try {
  if (fs.existsSync(BINARY_PATH)) {
    fs.unlinkSync(BINARY_PATH);
  }
  if (fs.existsSync(BIN_DIR)) {
    fs.rmdirSync(BIN_DIR);
  }
} catch (error) {
  console.error('Error during uninstall:', error);
  process.exit(1);
}

File: pip-package/shaper_pkg/uninstall.py
#!/usr/bin/env python3
# SPDX-License-Identifier: MPL-2.0

import sys
from pathlib import Path

# Get package directory
PACKAGE_DIR = Path(__file__).parent
BIN_DIR = PACKAGE_DIR / "bin"
BINARY_PATH = BIN_DIR / "shaper"

try:
    if BINARY_PATH.exists():
        BINARY_PATH.unlink()
    if BIN_DIR.exists() and not any(BIN_DIR.iterdir()):
        BIN_DIR.rmdir()
except Exception as error:
    print(f"Error during uninstall: {error}", file=sys.stderr)
    sys.exit(1)
